### PR TITLE
Tagged metrics API

### PR DESCRIPTION
**Pull request by chelseakomlo**
This PR:
- Adds an HTTP endpoint to access tagged metrics,
- Adds documentation for this HTTP endpoint
- Moves tagged metrics from client configuration to telemetry configuration
- Adds documentation for new telemetry config options

### CODE REVIEW

File: command/agent/config-test-fixtures/basic.hcl [hunk 1]
```diff
@@ -98,6 +96,8 @@ telemetry {
     collection_interval = "3s"
     publish_allocation_metrics = true
     publish_node_metrics = true
+  disable_tagged_metrics = true
```
Comment (dadgar) [inline command/agent/config-test-fixtures/basic.hcl]: Formatting

File: website/source/docs/agent/configuration/telemetry.html.md [hunk 2]
```diff
@@ -58,6 +58,14 @@ The following options are available on all telemetry configurations.
 - `publish_node_metrics` `(bool: false)` - Specifies if Nomad should publish
   runtime metrics of nodes.
 
+- `backwards_compatible_metrics` `(bool: false)` - Specifies if Nomad should
+  publish metrics that are backwards compatible with versions below 0.7, as
+  post version 0.7, Nomad emits tagged metrics.
+
+- `disable_tagged_metrics` `(bool: false)` - Specifies if Nomad should not emit
```
Comment (dadgar) [inline website/source/docs/agent/configuration/telemetry.html.md]: I would document that new metrics will only be added to tagged metrics. Further that these options are used to transition monitoring to tagged metrics and will eventually be deprecated.

File: website/source/docs/agent/configuration/telemetry.html.md [hunk 3]
```diff
@@ -162,3 +170,6 @@ These `telemetry` parameters apply to
   best use of this is to as a hint for which broker should be used based on
   *where* this particular instance is running (e.g. a specific geographic location or
   datacenter, dc:sfo).
+
```
Comment (dadgar) [inline website/source/docs/agent/configuration/telemetry.html.md]: Remove

File: website/source/api/metrics.html.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
+---
```
Comment (dadgar) [inline website/source/api/metrics.html.md]: What about this page: https://www.nomadproject.io/docs/agent/telemetry.html

File: client/client.go [hunk 5]
```diff
@@ -166,8 +169,15 @@ var (
 	noServersErr = errors.New("no servers")
 )
 
+// ClientTelemetry is a subset of global telemetry configuration options that
+// are relevant for the client
+type ClientTelemetry struct {
```
Comment (dadgar) [inline client/client.go]: Remove this and just add the telemetry block into the client configuration. Then the agent can just assign it when creating the client config. In the future client configs should go in the client/config/config.go

File: website/source/api/metrics.html.md (hunk 4 above)
Comment (chelseakomlo) [reply website/source/api/metrics.html.md]: See https://github.com/hashicorp/nomad/pull/3162/files#diff-9351a387bcf0a0b62128bd332eb78cb4R15

Do you think the metrics endpoint should be documented on https://www.nomadproject.io/docs/agent/telemetry.html ?

Comment (dadgar) [reply website/source/api/metrics.html.md]: I meant the values in the table

File: website/source/docs/agent/telemetry.html.md [hunk 6]
```diff
@@ -229,7 +236,196 @@ configuration block.
...
+    <th>Unit</th>
+    <th>Type</th>
+    <th>Labels</th>
+  </tr>
+  <tr>
+    <td>`nomad.client.allocated.cpu`</td>
+    <td>Total amount of CPU shares the scheduler has allocated to tasks</td>
+    <td>MHz</td>
+    <td>Gauge</td>
+    <td>Host ID</td>
```
Comment (dadgar) [inline website/source/docs/agent/telemetry.html.md]: Can you make these match the actual name: "node_id", "datacenter"

File: command/agent/agent_test.go [hunk 7]
```diff
@@ -316,6 +317,29 @@ func TestAgent_ClientConfig(t *testing.T) {
...
+	a := &Agent{config: conf}
+
+	c, err := a.clientConfig()
+	assert.Nil(err)
+
+	telemetry := conf.Telemetry
+
+	assert.Equal(c.StatsCollectionInterval, telemetry.collectionInterval)
+	assert.Equal(c.PublishNodeMetrics, telemetry.PublishNodeMetrics)
+	assert.Equal(c.PublishAllocationMetrics, telemetry.PublishAllocationMetrics)
```
Comment (dadgar) [inline command/agent/agent_test.go]: 👍

File: website/source/docs/agent/configuration/telemetry.html.md [hunk 8]
```diff
@@ -58,6 +58,20 @@ The following options are available on all telemetry configurations.
 - `publish_node_metrics` `(bool: false)` - Specifies if Nomad should publish
   runtime metrics of nodes.
 
+- `backwards_compatible_metrics` `(bool: false)` - Specifies if Nomad should
+  publish metrics that are backwards compatible with versions below 0.7, as
+  post version 0.7, Nomad emits tagged metrics. and all new metrics will
```
Comment (dadgar) [inline website/source/docs/agent/configuration/telemetry.html.md]: ". and all new".

Should be "tagged metrics. All new metrics will only be added to tagged metrics... ".

File: website/source/api/metrics.html.md (hunk 4 above)
Comment (chelseakomlo) [reply website/source/api/metrics.html.md]: Ah, understood. 

See https://github.com/hashicorp/nomad/pull/3162/commits/e5410d106c17ea57da5130981261cc5b97202dd6 , let me know if there is anything I can add/change with this.

**Pull request merged**